core/src/main/scala/chisel3/internal/firrtl/Converter.scala

```diff
@@ -351,7 +353,7 @@ private[chisel3] object Converter {
   }
 
   def convert(port: Port, topDir: SpecifiedDirection = SpecifiedDirection.Unspecified): fir.Port = {
-    val resolvedDir = SpecifiedDirection.fromParent(topDir, port.dir)
+    val resolvedDir = SpecifiedDirection.fromParent(topDir, firrtlUserDirOf(port.id))
     val dir = resolvedDir match {
```

Comment:
For reviewers, this slight change is the entire fix for Vecs, deleting the logic in RawModule is just a cleanup since that code is no longer necessary (it was not recursive and thus incomplete anyway).

The change to `firrtlUserDirOf` above combined with this change fixes the bug for `OpaqueTypes`.